Microsoft.AVS.Management/Microsoft.AVS.Management.psm1

```diff
@@ -270,6 +270,48 @@ function New-LDAPIdentitySource {
     }
 }
 
+<#
+    .Synopsis
+     Download the certificate from a domain controller
+#>
+function Get-CertificateFromDomainController {
+    param (
+        [Parameter(
+            Mandatory = $true)]
+        [ValidateNotNull()]
+        [System.Uri]
+        $ParsedUrl,
+
+        [Parameter(
+            Mandatory = $true)]
+        [ValidateNotNull()]
+        [string]
+        $computerUrl
+    )
+
+    try {
+        $Command = 'nslookup ' + $ParsedUrl.Host + ' -type=soa'
+        $SSHRes = Invoke-SSHCommand -Command $Command -SSHSession $SSH_Sessions['VC']
+        if ($SSHRes.ExitStatus -ne 0) { 
+            throw "The FQDN $ParsedUrl.Host cannot be resolved to an IP address. Make sure DNS is configured."
+        }
+
+        $Command = 'nc -vz ' + $ParsedUrl.Host + ' ' + $ParsedUrl.Port
+        $SSHRes = Invoke-SSHCommand -Command $Command -SSHSession $SSH_Sessions['VC']
+        if ($SSHRes.ExitStatus -ne 0) { 
+            throw "The connection cannot be established. Please check the address, routing and/or firewall and make sure the port $ParsedUrl.Port is open." 
```

Comment:
`make sure port $ParsedUrl.Port is open`

Reply:
Please try the string formatting in your pwsh console, I believe you need to enclose the expression like this for it to format correctly:
`$($ParsedUrl.Port)`